app/selectors/currencyRateController.ts

```diff
@@ -0,0 +1,38 @@
+import { createSelector } from 'reselect';
+import { EngineState } from './types';
+
+const selectCurrencyRateControllerState = (state: EngineState) => ({
+  conversionRate:
+    state?.engine?.backgroundState?.CurrencyRateController?.conversionRate,
+  currentCurrency:
+    state?.engine?.backgroundState?.CurrencyRateController?.currentCurrency,
+  nativeCurrency:
+    state?.engine?.backgroundState?.CurrencyRateController?.nativeCurrency,
+});
```

Comment:
Nit: Perhaps it would be simpler to return the entire controller state here, rather than hardcode each property?
```suggestion
const selectCurrencyRateControllerState = (state: EngineState) =>
  state?.engine?.backgroundState?.CurrencyRateController;
```